src/mam/mod_mam_rdbms_arch.erl

```diff
@@ -340,12 +338,13 @@ prepare_insert(Name, NumRows) ->
     ok.
 
 %% Removal logic
--spec remove_archive(Acc :: mongoose_acc:t(), HostType :: host_type(),
-                     ArcID :: mod_mam:archive_id(),
-                     RoomJID :: jid:jid()) -> mongoose_acc:t().
-remove_archive(Acc, HostType, ArcID, _ArcJID) ->
+-spec remove_archive(Acc, Params, Extra) -> {ok, Acc} when
+    Acc :: term(),
+    Params :: map(),
```

Comment:
All these params should also not lose type information, they could be like `#{archive_id := mod_mam:archive_id()}`. That is useful even if only when debugging this information, it already tells the reader what are we handling.